interfaces/builtin/mount_control.go

```diff
@@ -0,0 +1,337 @@
+	var list []string
+	value, ok := mountEntry[fieldName]
+	if ok {
+		interfaceList, ok := value.([]interface{})
+		if !ok {
+			return nil, fmt.Errorf(`mount-control "mount.%s" must be an array of strings (got %q)`, fieldName, value)
+		}
+		for _, iface := range interfaceList {
+			valueString, ok := iface.(string)
+			if !ok {
+				return nil, fmt.Errorf(`mount-control "mount.%s" element not a string (%q)`, fieldName, iface)
+			}
+			list = append(list, valueString)
+		}
+	}
+	return list, nil
+}
+
+func enumerateMounts(plug interfaces.Attrer, fn func(mountInfo *MountInfo) error) error {
+	mountAttr, ok := plug.Lookup("mount")
+	if !ok {
+		return nil
+	}
+	mounts, ok := mountAttr.([]interface{})
+	if !ok {
+		return errors.New(mountAttrNotADictionary)
+	}
+
+	for _, m := range mounts {
+		mount, ok := m.(map[string]interface{})
+		if !ok {
+			return errors.New(mountAttrNotADictionary)
+		}
+
+		what, ok := mount["what"].(string)
+		if !ok {
+			return fmt.Errorf(`mount-control "mount.what" must be a string`)
+		}
+
+		where, ok := mount["where"].(string)
+		if !ok {
+			return fmt.Errorf(`mount-control "mount.where" must be a string`)
+		}
+
+		persistent := false
+		persistentValue, ok := mount["persistent"]
+		if ok {
+			if persistent, ok = persistentValue.(bool); !ok {
+				return fmt.Errorf(`mount-control "mount.persistent" must be a boolean`)
+			}
+		}
+
+		types, err := parseStringList(mount, "type")
+		if err != nil {
+			return err
+		}
+
+		options, err := parseStringList(mount, "options")
+		if err != nil {
+			return err
+		}
+
+		mountInfo := &MountInfo{
+			what:       what,
+			where:      where,
+			persistent: persistent,
+			types:      types,
+			options:    options,
+		}
+
+		if err := fn(mountInfo); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateWhatAttr(what string) error {
+	if what[0] != '/' {
+		return fmt.Errorf(`mount-control "what" attribute must be an absolute path`)
+	}
+
+	if !cleanSubPath(what) {
+		return fmt.Errorf(`mount-control "what" pattern is not clean: %q`, what)
+	}
+
+	if _, err := utils.NewPathPattern(what); err != nil {
+		return fmt.Errorf(`mount-control "what" attribute error: %v`, err)
+	}
+
+	return nil
+}
+
+func validateWhereAttr(where string) error {
+	if !whereRegexp.MatchString(where) {
+		return fmt.Errorf(`mount-control "where" attribute is invalid`)
+	}
+
+	if !cleanSubPath(where) {
+		return fmt.Errorf(`mount-control "where" pattern is not clean: %q`, where)
+	}
+
+	if _, err := utils.NewPathPattern(where); err != nil {
+		return fmt.Errorf(`mount-control "where" attribute error: %v`, err)
+	}
+
+	return nil
+}
+
+func validateMountTypes(types []string) error {
+	for _, t := range types {
+		if !typeRegexp.MatchString(t) {
+			return fmt.Errorf(`mount-control filesystem type invalid: %q`, t)
+		}
+	}
+	return nil
+}
+
+func validateMountOptions(options []string) error {
+	for _, o := range options {
```

Comment:
nitpick^100

you could potentially use strutil.Intersection here with options and allowedMountOptions, then if the length of the intersection is smaller than the size of `options`, there are unsupported options, but that's pretty much exactly what this code is doing anyways

Reply:
yeah that's totally fair, this implementation is completely fine as-is